### PR TITLE
add support for endpoint parameter in cluster_config

### PR DESCRIPTION
**Pull request by mihivagyok**
Adding new parameter `endpoint_type` for cluster_config datasource (IBM Cloud Kubernetes Service). This parameter can set what kind of URL is generated into the kubeconfig. The default is public - if the cluster has public service endpoint enabled option.

Possible options: 
- empty
- private (works for classic and VPC infrastructure)
- vpe (only for VPC)
- link (only for Satellite)

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

changed tests:
```
$ make testacc TESTARGS='-run=TestAccIBMContainer_ClusterConfigCalicoDataSourceBasic'
--- PASS: TestAccIBMContainer_ClusterConfigCalicoDataSourceBasic (1074.02s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes      1075.088s
```
default tests are passing too:
```
=== RUN   TestAccIBMContainer_ClusterConfigDataSourceBasic
--- PASS: TestAccIBMContainer_ClusterConfigDataSourceBasic (1315.53s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes      1316.606s
...
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -228,6 +228,8 @@ replace github.com/dgrijalva/jwt-go v3.2.0+incompatible => github.com/golang-jwt
 
 replace github.com/portworx/sched-ops v0.0.0-20200831185134-3e8010dc7056 => github.com/portworx/sched-ops v0.20.4-openstorage-rc3 // required by rook v1.7
 
+replace github.com/IBM-Cloud/bluemix-go v0.0.0-20230601050310-eecebfbff63e => github.com/mihivagyok/bluemix-go v0.0.0-20230908132747-880981f4c25b
```
Comment (mihivagyok) [inline go.mod:231]: I will remove this once the bluemix-go PR is merged.
https://github.com/IBM-Cloud/bluemix-go/pull/395

File: ibm/service/kubernetes/data_source_ibm_container_cluster_config.go [hunk 2]
```diff
@@ -88,6 +88,12 @@ func DataSourceIBMContainerClusterConfig() *schema.Resource {
 				Optional:    true,
 				Default:     false,
 			},
+			"endpoint_type": {
+				Description: "It can specify what kind of server URL will be used for the cluster context",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     false,
```
Comment (hkantare) [inline ibm/service/kubernetes/data_source_ibm_container_cluster_config.go:95]: How can we have Default value as "false" as we know the supported are "private", "link", "vpe"

If we don't have any Default value how does the SDK handles

Comment (mihivagyok) [reply ibm/service/kubernetes/data_source_ibm_container_cluster_config.go:95]: Yes, this is my bad. The default shall be empty string. If it is empty, the SDK won't set he `endpoint` field in the request. Thanks!

Comment (hkantare) [reply ibm/service/kubernetes/data_source_ibm_container_cluster_config.go:95]: We don't need to explictly set Default = "" by default in Terraform string will be empty

Comment (mihivagyok) [reply ibm/service/kubernetes/data_source_ibm_container_cluster_config.go]: Yes, I have removed this whole line! Thanks!

**Comment by hkantare** [comment]
Doc also need to be updated with new argument websites/docs/d

**Comment by mihivagyok** [comment]
> Doc also need to be updated with new argument websites/docs/d

Yes, I will update it. I wanted to ask where to add these changes, so thank you very much!

**Pull request merged**